### PR TITLE
Optional Setup Scripts

### PR DESCRIPTION
**Pull request by matthewdfuller**
## Changelog
* Added checks in `init` for leader and follower setup scripts in /opt services

## Notes
No Docker runs

### CODE REVIEW

**Comment by ts3ng** [comment]
thanks for adding this @matthewdfuller  okay this looks fine to me so my PR will include scripts in splunk/common.

File: init [hunk 1]
```diff
@@ -37,6 +37,20 @@ function leader-setup() {
...
   done
 
+  for service in $(etcd-get /environment/services)
+  do
+      servicedir=${SCRIPTDIR}/${VERSION}/opt/${service}
+      if [[ ! -d $servicedir ]]; then
+          continue
+      fi
+
+      for script in $(ls $servicedir/leader|grep -e '.sh$')
```
Comment (chr0n1x) [inline init]: does it matter if the `leader` scripts happen first or the `opt`? Is this order something that you always want to guarantee?

**Comment by chr0n1x** [comment]
👍

**Comment by chr0n1x** [comment]
👍

**Pull request merged**